src/torch_onnx/_patch.py

```diff
@@ -48,41 +48,48 @@ def _from_dynamic_axes_to_dynamic_shapes(
     """
     # https://github.com/pytorch/pytorch/pull/128371
     if dynamic_axes is None:
-        return None
+        return None, input_names
 
-    input_names_set = set() if input_names is None else set(input_names)
-
-    dynamic_shapes = {}
+    if input_names is not None:
+        sig = _signature(model)
+        if len(input_names) > len(sig.parameters):
+            raise ValueError(
+                f"Number of input names ({len(input_names)}) should not be greater than the number of model inputs ({len(sig.parameters)})"
+            )
+        input_names_to_model_inputs = {
+            input_name: param_name
+            for input_name, param_name in zip(input_names, sig.parameters)
+        }
+
+    # NOTE: torch.export.export does not support input names assignmen,
+    # so we need to map input names to model inputs to create dynamic_shapes
+    # for the exported program
+    dynamic_shapesto_exported_program = {}
     for input_name, axes in dynamic_axes.items():
-        if input_name in input_names_set:
+        if input_name not in input_names_to_model_inputs:
             raise ValueError(
-                "input names is not supported yet. Please use model forward signature."
+                f"dynamix axe: {input_name} is not found in the input names: {input_names}"
             )
+        model_input_name = input_names_to_model_inputs[input_name]
```

Comment:
or model input is the name presented in the forward function right??

Reply:
Yes